x/mm2/keeper/genesis_test.go

```diff
@@ -0,0 +1,113 @@
+package keeper_test
+
+import (
+	sdkmath "cosmossdk.io/math"
+
+	"github.com/skip-mev/slinky/x/mm2/types"
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+)
+
+func (s *KeeperTestSuite) TestInitGenesisInvalidGenesis() {
+	s.Run("test that init genesis with invalid genesis params - fails", func() {
+		// create a fake genesis state with invalid params
+		gs := types.GenesisState{
+			MarketMap: types.DefaultGenesisState().MarketMap,
+			Params: types.Params{
+				MarketAuthorities: []string{"invalid"},
+			},
+		}
+
+		// assert that InitGenesis panics
+		s.Require().Panics(func() {
+			s.keeper.InitGenesis(s.ctx, gs)
+		})
+	})
+
+	s.Run("test that init genesis with invalid duplicate runs - fails", func() {
+		// create a valid genesis
+		gs := types.DefaultGenesisState()
+
+		gs.MarketMap = types.MarketMap{
+			Markets: map[string]types.Market{
+				ethusdt.Ticker.String(): ethusdt,
+				btcusdt.Ticker.String(): btcusdt,
+				usdcusd.Ticker.String(): usdcusd,
+			},
+		}
+
+		// assert that InitGenesis panics
+		s.Require().Panics(func() {
+			s.keeper.InitGenesis(s.ctx, *gs)
+			s.keeper.InitGenesis(s.ctx, *gs)
+		})
+	})
```

Comment:
What is this testing?

Reply:
Can we add a test that checks that the market map is correctly configured? i.e. normalizeByPair exists for all etc.

Reply:
we can't run init genesis twice.  this is because the markets are already set in state and you can't do that again